### PR TITLE
Added FAQ model, routes, and controller draft

### PR DESCRIPTION
**Pull request by spiray**
I added a draft of the FAQ structure. Please provide feedback if this is generally what we are looking for.

### CODE REVIEW

**Comment by spiray** [comment]
[Create FAQ Model](https://trello.com/c/l2bRtx8X/62-create-faq-model)

File: src/api/faq/faq.controller.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  answer: 'An inexpensive mechanical keyboard from Amazon. Check it out here: https://amzn.to/2EwYmSd',
+},
+{
+  id: 5,
+  question: 'How long have you been coding?',
+  answer: 'Over 15 years! I started coding HTML / CSS websites as a kid. Learned Java, C, C++ in college. Wrote C# / .NET desktop applications for a while. Started learning modern web technologies in my spare time, and taught JavaScript full stack web development for 3+ years.',
+},
+{
+  id: 6,
+  question: 'What\'s the best way to contact you ?',
```
Comment (ewenjo) [inline src/api/faq/faq.controller.js]: Remove space before question mark.

File: src/api/faq/faq.controller.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  answer: 'Join the discord. https://coding.garden/discord',
+},
+{
+  id: 7,
+  question: 'How do you add emojis in VS Code?',
+  answer: 'This is a feature of Mac OS X. Press CTRL+CMD+Space to bring up the emoji menu! On Windows 10 you can use CTRL+Period  ',
```
Comment (ewenjo) [inline src/api/faq/faq.controller.js]: Remove extra spaces at the end.

**Review by ewenjo — CHANGES_REQUESTED** [review]
Looks good! Tested, and everything works fine.

Just a few minor spellings mistakes. See comments.

File: src/api/faq/faq.controller.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+{
+  id: 7,
+  question: 'How do you add emojis in VS Code?',
+  answer: 'This is a feature of Mac OS X. Press CTRL+CMD+Space to bring up the emoji menu! On Windows 10 you can use CTRL+Period',
+},
+];
+
+function getFAQ(req, res) {
+  const response = 'id' in req.params ? [faqData.find(({ id }) => id === Number(req.params.id))]
+    : faqData;
```
Comment (ewenjo) [inline src/api/faq/faq.controller.js:45]: Weird formatting in my opinion, but shouldn't matter that much.

**Review by ewenjo — APPROVED** [review]
Good work!

File: src/api/faq/faq.controller.js (hunk 3 above)
Comment (spiray) [reply src/api/faq/faq.controller.js:45]: I considered many different ways. There are tens of ways to do this, which way do you prefer?

Comment (ewenjo) [reply src/api/faq/faq.controller.js]: I liked the way you had it before.

`
const response = 'id' in req.params 
? [faqData.find(({ id }) => id === Number(req.params.id))]
: faqData;
`

123

Comment (ewenjo) [reply src/api/faq/faq.controller.js:45]: I like the way you did it here. ba21d98dbf4c9d3e1a859062b74027172cdd73a6

**Comment by rabbitwerks** [comment]
Merging. Api may be updated at a later point, if we determine custom api architecture is more effective than standardized.

**Review by rabbitwerks — APPROVED** [review]

**Pull request merged**